packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts

```diff
@@ -383,16 +408,72 @@ function querySelectorAllPatched(this: Element /*, selector: string*/): NodeList
     } else if (isNodeShadowed(this)) {
         // element inside a shadowRoot
         const ownerKey = getNodeOwnerKey(this);
-        filtered = ArrayFilter.call(nodeList, elm => getNodeOwnerKey(elm) === ownerKey);
+        if (!isUndefined(ownerKey) || ENABLE_ELEMENT_PATCH) {
+            // The patch is enabled or `this` is an element rendered by lwc
+            filtered = ArrayFilter.call(nodeList, elm => getNodeOwnerKey(elm) === ownerKey);
+        } else {
+            // `this` is a manually inserted element inside a shadowRoot
+            filtered = ArraySlice.call(nodeList);
+        }
     } else {
-        // element belonging to the document
-        filtered = ArrayFilter.call(
-            nodeList,
-            // TODO: issue #1222 - remove global bypass
-            elm => isUndefined(getNodeOwnerKey(elm)) || isGlobalPatchingSkipped(this)
-        );
+        if (this instanceof HTMLBodyElement || ENABLE_ELEMENT_PATCH) {
+            // `this` is document.body or element belonging to the document with the patch enabled
+            filtered = ArrayFilter.call(
+                nodeList,
+                // TODO: issue #1222 - remove global bypass
+                elm => isUndefined(getNodeOwnerKey(elm)) || isGlobalPatchingSkipped(this)
+            );
+        } else {
+            // `this` is outside the lwc boundary and patch is not enabled.
+            filtered = ArraySlice.call(nodeList);
+        }
     }
-    return createStaticNodeList(filtered);
+    return filtered;
+}
+
+function querySelectorAllPatched(this: Element /*, selector: string*/): NodeListOf<Element> {
+    const nodeList = arrayFromCollection(
+        elementQuerySelectorAll.apply(this, ArraySlice.call(arguments) as [string])
+    );
+    const filteredResults = filterNodeListQueryResult.call(this, nodeList);
+
+    return createStaticNodeList(filteredResults);
+}
+
+function getElementsByTagNamePatched(
+    this: Element /*, tagName: string*/
+): HTMLCollectionOf<Element> {
+    const elements = arrayFromCollection(
```

Comment:
@caridy @ekashida `getElementsByTagName` is not patched, even inside lwc boundary, (`this.template.getElementsByTagName` pierce through shadowRoot) this logic is fixing that. Do we want to also put that behind a flag?